flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/codegen/calls/ScalarFunctions.scala

```diff
@@ -143,15 +143,30 @@ object ScalarFunctions {
     new MultiTypeMethodCallGen(BuiltInMethods.ABS))
 
   addSqlFunction(
+    ABS,
+    Seq(BIG_DEC_TYPE_INFO),
+    new MultiTypeMethodCallGen(BuiltInMethods.ABS_DEC))
+
+  addSqlFunction(
```

Comment:
`FLOAT`, `INT` etc. is supported implicitly as they `shouldAutocastTo` `DOUBLE`. The most suitable method is later chosen by the Java compiler. Calcite's `SqlFunctions` provides methods for all basic types.